### PR TITLE
Slow down before elevator

### PR DESCRIPTION
**Pull request by Sajdkick**
It's a bit weird. I wanted to slow the cart down in the same way we do in the other sections, the problem is that in the elevator scene, we assume that we stop when hitting a trigger. And the script for slowing down stops after a set distance. So if the cart stops to early, we won't hit the trigger. But if we stop to late it won't be smooth. So i had to use trial and error to find the precise distance where it slows down and stops on the trigger. (this distance happened to be 13.37 lul). So it's a bit weird. But it works.

### CODE REVIEW

File: Game/Resources/elevatorScene/Scripts/elevator_moveCart.as [hunk 1]
```diff
@@ -67,15 +76,39 @@ class elevator_moveCart {
...
+                if (stopTime < endTime){
+                    stopTime += deltaTime;
+                    tempPos = self.GetWorldPosition();
+                    float s = a * stopTime * stopTime * stopTime / 3 + b * stopTime * stopTime / 2 + c * stopTime;
+                    tempPos.x = startStopPos - s;
+                    self.SetWorldPosition(tempPos);
+                }
+                break;
+            case 4:
+                print("4");
```
Comment (andens) [inline Game/Resources/elevatorScene/Scripts/elevator_moveCart.as]: I guess this is a waiting phase, but we probably shouldn't print alot of stuff when in this state.

Comment (Sajdkick) [reply Game/Resources/elevatorScene/Scripts/elevator_moveCart.as]: Ahhh forgot to remove this debugging code!

**Comment by andens** [comment]
Come to think of it, I think the triggers are there to make the cart stop at a proper location. The braking code knows when we have arrived, at which point we can do whatever the triggers would normally do. That way we don't have to tweak things so that we hit them precisely. It's still important that the cart end up in a proper location, but I found a nice solution for that in the voltage puzzle. What I did was to place an entity without components where I wanted the cart to stop, then I use its position when calculating `brakeDistance`. That way I could easily just move the trigger volume to not have the cart increase velocity before braking, and I end up exactly where I want to :)

**Comment by Sajdkick** [comment]
@andens Yeah it shouldn't be too hard to remove the trigger, but the question is if we want to meddle with it, or do something else that's higher priority.

**Comment by Sajdkick** [comment]
I removed the triggers and stuff now, so it should be smooth. I can't get the doors to close behind the player though. Even though i call CloseDoors(). Maybe it's supposed to be like that?

**Comment by andens** [comment]
Bump @marredal

**Comment by andens** [comment]
Has this been tested? The trigger that called `StopCart()` has been removed, but I can't see any replacement for this? Considering that this method eventually sends a message to the board it seems like a breaking change if it's not called.

**Comment by Sajdkick** [comment]
I replaced the trigger with some code in the minecart script. But i only close the doors, so i've missed the board. I fix.

**Comment by Sajdkick** [comment]
Wait it actually does work. In elevator_moveCart -> Update -> phase == 3 we switch to 4 when we've stopped. And this signals the board.

**Comment by marredal** [comment]
cage.GetSoundSource().Stop() was called upon in StopCart() but if the trigger is gone you should move this into phase 4.

**Pull request merged**